### PR TITLE
OCD-3278 - Require D12 and D13 on new certified products with certain criteria

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-3278

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dto/listing/pending/PendingCertifiedProductDTO.java [hunk 1]
```diff
@@ -65,8 +73,10 @@
     private Long productClassificationId;
     private Boolean deleted;
     private Long lastModifiedUser;
-    private Set<String> errorMessages;
-    private Set<String> warningMessages;
+    @Builder.Default
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/dto/listing/pending/PendingCertifiedProductDTO.java]: Does this work? I tried something like this with something, and got an NPE when trying to add to this array

File: chpl/chpl-resources/src/main/resources/environment.properties [hunk 2]
```diff
@@ -130,3 +130,10 @@ schemaCsv2015NameLegacy=2015 Listing CSV Data Dictionary Legacy.csv
 schemaSurveillanceName=Surveillance and Nonconformity CSV Data Dictionary.csv
 schemaBasicSurveillanceName=Surveillance (basic) CSV Data Dictionary.csv
 #############################################################
+
+######  CRITERIA THAT REQUIRE 170.315 (d)(12) and 170.315 (d)(13)  ######
+requiredCriteriaDependencies=[{\
+    "dependentCriteria": [1,2,3,4,5,9,10,12,13,14,15,16,169,17,170,18,171,21,22,172,23,73,24,174,25,26,27,175,28,30,31,38,40,176,41,42,43,44,45,46,47,177,48,49,56,57,58,179,59,60],\
```
Comment (andlar) [inline chpl/chpl-resources/src/main/resources/environment.properties]: Do these IDs include _all_ of the new ones? Or, at least, count all of them?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/dto/listing/pending/PendingCertifiedProductDTO.java (hunk 1 above)
Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/dto/listing/pending/PendingCertifiedProductDTO.java]: Same question. This was originally added in a ticket for OCD-3245 and is in what I merged into qa this morning during the reset.

File: chpl/chpl-resources/src/main/resources/environment.properties (hunk 2 above)
Comment (andlar) [reply chpl/chpl-resources/src/main/resources/environment.properties]: Or maybe is this also blocked by OCD-3283?

Comment (tmy1313) [reply chpl/chpl-resources/src/main/resources/environment.properties]: D2, D3, D10 are included in this list.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/util/ValidationUtils.java [hunk 3]
```diff
@@ -197,15 +213,21 @@ public static CertificationCriterion getCert(String certNumber, List<Certificati
     }
 
     /**
-     * Look for required complimentary criteria; if any one of the
-     * criterionToCheck is present in allCriteriaMet then all of the
-     * complimentaryCertNumbers must be present in allCriteriaMet.
+     * DEPLRECATED - Look for required complimentary criteria; if any one of the criterionToCheck is present in
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/util/ValidationUtils.java]: The only actual change in this commit is adding the `@Deprecated` tag to this method, right? Can we just delete all of the javadoc stuff? Do they actually provide value? I'm seeing so many changes that are literally just whitespace changes I'd rather just have these things removed

**Pull request merged**